### DISCUSSION TITLE
Fprime Unit Testing - Branch Coverage

### DISCUSSION THREAD

**Discussion by arturodrt** [nasa/fprime#1858] [Q&A] [answered]
Hello, 

does anyone know how I can achieve a higher level of coverage for my unit test, since the gcov tool that checks for code coverage indicates that I am missing a branch in the following code statement? Right now it is not clear to me how to cover the second branch that the report indicates.

![branch_coverage](https://user-images.githubusercontent.com/63877617/214086874-bd1faafe-b6c9-4a77-bab0-6964424db89a.png)

**Comment by LeStarch** [comment, marked as answer]
This is discussed over on `gcovr`'s FAQ: https://gcovr.com/en/stable/faq.html#why-does-c-code-have-so-many-uncovered-branches.  I haven't looked to deeply as we typically look at line coverage.  However, it it good to understand the data.  Thanks for noting this.

**Reply by LeStarch** [reply]
I should also note: when running coverage through `fprime-util coverage` you can supply flags to `gcovr` via the pass through flag.  
Of the top of my head it is:

```
fprime-util coverage --pass-through --some-gcovr-flag
```
However, it may be worth looking at the `--help` flag to check I got that right.
